src/Workspaces/Core/Portable/Workspace/Solution/SourceGeneratorIdentity.cs

```diff
@@ -8,16 +8,16 @@
 
 namespace Microsoft.CodeAnalysis
 {
-    internal record struct SourceGeneratorIdentity(string AssemblyName, string TypeName)
+    internal record struct SourceGeneratorIdentity(string AssemblyName, Version AssemblyVersion, string TypeName)
     {
         public SourceGeneratorIdentity(ISourceGenerator generator)
-            : this(GetGeneratorAssemblyName(generator), GetGeneratorTypeName(generator))
+            : this(GetGeneratorAssemblyName(generator), generator.GetGeneratorType().Assembly.GetName().Version!, GetGeneratorTypeName(generator))
         {
         }
 
         public static string GetGeneratorAssemblyName(ISourceGenerator generator)
         {
-            return generator.GetGeneratorType().Assembly.FullName!;
+            return generator.GetGeneratorType().Assembly.GetName().Name!;
```

Comment:
curious why?

Reply:
The full name is the form "MySourceGenerator, Version=1.2.3.4"; which means we'd be capturing the version implicitly in the name. It didn't really matter until now, when we want those reported as separate telemetry points.